### PR TITLE
Update `eth-snap-keyring` to 0.2.x to use the `keyring-api` 0.2.x

### PR DESCRIPTION
**Pull request by danroc**
## Explanation

This PR updates the version of the `eth-snap-keyring` to use the new version of the `keyring-api` that we plan to expose to snap developers.

This PR has no other impacts, and only affects Flask.

**BREAKING CHANGE**: The `keyring-api` 0.2.x is incompatible with the previous 0.1.x release. Snaps developed with version 0.1.x of the API need to migrate to 0.2.x.

## Manual Testing Steps

1. Install the SSK from the [SSK dapp][snap-dapp].
2. Create an account using the [SSK dapp][snap-dapp].
    1. Note that the account name defined in the SSK dapp will be ignored.
3. Use the [test dapp](https://metamask.github.io/test-dapp/) to sign transactions using the sync and async flows.
    1. There is a toggle in the [SSK dapp][snap-dapp] to switch between the flows.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

[snap-dapp]: https://metamask.github.io/snap-simple-keyring/0.2.2/

### CODE REVIEW

**Comment by danroc** [comment]
@metamaskbot update-policies

**Comment by HowardBraham** [comment]
It's not working for me, when I try to Create Account I get this error:

`{"code":-32603,"message":"Method not supported: createAccount","data":{"originalError":{}}}`

**Comment by danroc** [comment]
Indeed @HowardBraham, I'll publish the new SSK that works with this branch (0.2.1).

**Review by gantunesr — DISMISSED** [review]

**Review by FrederikBolding — DISMISSED** [review]

**Review by gantunesr — DISMISSED** [review]

**Review by HowardBraham — DISMISSED** [review]

**Comment by plasmacorral** [comment]
I am having issues signing or sending with the test dapp when I am using a snap account, on both a local ganache instance on chain_id=1, as well as Goerli.

The following images were from attempting to sign with "Sign Permit" in the test dapp.

Left image is with the snap account connected to the test dapp, right is an imported account and is able to sign succesfully:
<div style="display: inline-block"> </div>

Note that the following parameters at index=1 are absent when I am using a snap account:
`{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"verifyingContract","type":"address"},{"name":"chainId","type":"uint256"}],"Permit":[{"name":"owner","type":"address"},{"name":"spender","type":"address"},{"name":"value","type":"uint256"},{"name":"nonce","type":"uint256"},{"name":"deadline","type":"uint256"}]},"primaryType":"Permit","domain":{"name":"MyToken","version":"1","verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC","chainId":1},"message":{"owner":"0x9971d4b9917f26ac5c65292d207e6fe75cac6cca","spender":"0x5B38Da6a701c568545dCfcB03FcB875f56beddC4","value":3000,"nonce":0,"deadline":50000000000}}`

**Comment by plasmacorral** [comment]
Here are errors presented when I try to complete a send with a snap account.  Right is console from the test dapp and left is extension

**Comment by danroc** [comment]
@plasmacorral I cannot reproduce on my side, are you using the snap in the sync or async mode?

https://github.com/MetaMask/metamask-extension/assets/68558152/077f46ff-bef7-4f52-959e-4464762fdc62

**Comment by plasmacorral** [comment]
@danroc
Screenshot

File: lavamoat/browserify/flask/policy.json [hunk 1]
```diff
@@ -997,7 +982,7 @@
         "@metamask/eth-json-rpc-middleware>clone": true,
         "@metamask/eth-json-rpc-middleware>pify": true,
         "@metamask/eth-json-rpc-middleware>safe-stable-stringify": true,
-        "@metamask/eth-snap-keyring>@metamask/eth-sig-util": true,
+        "@metamask/eth-trezor-keyring>@metamask/eth-sig-util": true,
         "eth-rpc-errors": true,
```
Comment (gantunesr) [inline lavamoat/browserify/flask/policy.json:986]: @legobeat why are these changes occurring when updating the policy files? This PR should have any impact in the trezor keyring

Comment (gantunesr) [reply lavamoat/browserify/flask/policy.json:986]: Thanks for the explanation @legobeat

**Comment by danroc** [comment]
@plasmacorral can you please test again? We released the SSK 0.2.2 which should fix the toggle problem.

**Review by gantunesr — DISMISSED** [review]

**Review by HowardBraham — DISMISSED** [review]

**Comment by danroc** [comment]
@legobeat, the difference in the dependencies doesn't seem to affect us (did some manual tests). Should we do something else?

**Review by weizman — DISMISSED** [review]
approving the LM policies

**Review by FrederikBolding — DISMISSED** [review]

**Comment by FrederikBolding** [comment]
@legobeat I am working on removing the `-flask` versions of the snaps packages, this should start to make the dependency management much easier.

**Comment by plasmacorral** [comment]
Manually tested on Mac Ventura 13.5.2 primarily with Chrome 117, but also Firefox 117.

- Confirmed happy path snap install and account creation, both sending and signing transactions in both sync and async mode with a Snap created account from SSK v0.2.3.  
- Confirmed with multiple versions of SSK using multiple sets of accounts created by multiple snap installations that a user can send/sign from either set of accounts with the test dapp.  
- Accounts created in SSK 0.2.2 remain usable for sign/send after update to v0.2.3

**Comment by danroc** [comment]
@metamaskbot update-policies

**Comment by danroc** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
Wondering if [my PR here](https://github.com/MetaMask/eth-snap-keyring/pull/102) relates to this one at all

**Comment by danroc** [comment]
Yes @jiexi, we will soon release a new version of the `eth-snap-keyring` (including your PR), and update the extension again.

**Review by gantunesr — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**